xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -3939,9 +3940,13 @@ public int createUser(boolean withValidation, String userRights, XWikiContext co
                 }
             }
 
-            if ((!password.equals(password2))) {
+            ResetPasswordManager resetPasswordManager = Utils.getComponent(ResetPasswordManager.class);
+            if (!password.equals(password2)) {
                 // TODO: throw wrong password exception
                 return -2;
+            } else if (!StringUtils.isEmpty(password) &&
```

Comment:
@tmortagne would be good to have a confirmation this cannot mess up wikis with external authentication mechanism.

Reply:
All custom authenticators relaying on an external identity provider (LDAP, OIDC, SAML, etc) create users with an empty password by definition. That being said I guess it's possible that exist a custom authenticator which does set a password, and it might not like too much being forced to follow the default rules.

To be honest XWiki#createUser feels too low level to have a check like this.

Reply:
> To be honest, XWiki#createUser feels too low level to have a check like this.

Not sure where it should go then?

Reply:
@tmortagne so I might actually just get rid of that check for now: we now have inline registration using also livevalidation, so with JS this is covered (and tested by integration test). Without JS it's not covered, but it has never been so no regression. IMO we should plan having this kind of check when we'll move the checks done on the form in the backend from the register_macros.vm to a proper Java API which I'm not planning to do right now.